ui/src/app/pages/setting/component/asset-edit/asset-edit.component.ts

```diff
@@ -66,32 +182,41 @@ export class AssetEditComponent implements OnInit {
   }
   save(){
 
+    let asset:AssetModule = new AssetModule();
     if(this.advance == true){
-      let assetobj = JSON.parse(this.assetjsondata);
-      if(assetobj.id != this.asset.id){
-        this.addfail = true;
+      let jsondata = this.assetjsondata;
+      asset = JSON.parse(jsondata);
+
+      if(asset.id != this.asset.id){
+        this.editfail = true;
         this.tip = "Don't change asset ID.";
         return;
       }
-      this.asset = JSON.parse(this.assetjsondata);
+    }else{
+      asset= this.editAssetForm.value;
+      if(asset.category == 'Sensors'){
+        let realtimespec:AssetRealtimeDataSpecModule = new AssetRealtimeDataSpecModule();
+        realtimespec.unit = this.editAssetForm.get('unit').value;
+        realtimespec.validNumMax = this.editAssetForm.get('validNumMax').value;
+        realtimespec.validNumMin = this.editAssetForm.get('validNumMin').value;
+        asset.assetRealtimeDataSpec = realtimespec;
+      }
+      this.editassetloading = true;
+      asset.assetSource = "flowgate";
+      let assetStatus:AssetStatusModule = new AssetStatusModule();
+      assetStatus.status = this.editAssetForm.get('statusDisplay').value;
+      asset.status = assetStatus
     }
-    
-    this.editassetloading = true;
-    this.service.updateAssetsByID(this.asset).subscribe(
+
+    this.service.createAnAsset(asset).subscribe(
       (data)=>{
         this.editassetloading = false;
         this.router.navigate(["/ui/nav/setting/asset-list"]);
-      },
-      error=>{
+      },error=>{
         this.editassetloading = false;
```

Comment:
If you need to use the  error response, you should  specify the type. error ===>(error:HttpErrorResponse)